From 3f994d8acac5849363b0965a30a37ade77986b06 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Tue, 9 Jan 2018 16:01:49 +0200
Subject: [PATCH 3177/5242] MLK-17355-1: ASoC: fsl_mqs: Fix potential
 uninitialized pointer read

commit  22179024cdca896a116ab68cb2ae3cb01bc22d1f from
https://source.codeaurora.org/external/imx/linux-imx.git

Initialize gpr_np in order to avoid potential unitialized
pointer read in the section following the "out:" label.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/fsl_mqs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/fsl_mqs.c b/sound/soc/codecs/fsl_mqs.c
index 0753fac..402923f 100644
--- a/sound/soc/codecs/fsl_mqs.c
+++ b/sound/soc/codecs/fsl_mqs.c
@@ -204,7 +204,7 @@ static void fsl_mqs_shutdown(struct snd_pcm_substream *substream,
 static int fsl_mqs_probe(struct platform_device *pdev)
 {
 	struct device_node *np = pdev->dev.of_node;
-	struct device_node *gpr_np;
+	struct device_node *gpr_np = 0;
 	struct fsl_mqs *mqs_priv;
 	struct resource *res;
 	void __iomem *regs;
-- 
1.7.9.5

